srv/modules/runners/ui_rgw.py

```diff
@@ -0,0 +1,122 @@
+                                'secret_key': None,
+                                'urls': []}
+            self.pathname = pathname
+            self._admin()
+            self._urls()
+
+    def _canned(self, canned):
+        """
+        Return examples for debugging without a working Ceph cluster
+        """
+        if canned == 1:
+            self.credentials = {'access_key': "ABCDEFGHIJKLMNOPQRST",
+                                'secret_key': "0123456789012345678901234567890123456789",
+                                'urls': ["http://rgw1"]}
+        elif canned == 2:
+            self.credentials = {'access_key': "ABCDEFGHIJKLMNOPQRST",
+                                'secret_key': "0123456789012345678901234567890123456789",
+                                'urls': ["http://red1",
+                                         "http://red2",
+                                         "http://blue1:8000",
+                                         "http://blue2:8000"]}
+
+    def _admin(self, filename="user.admin.json"):
+        """
+        Expect admin user file; otherwise, search for first system user.
+        Update access_key, secret_key
+        """
+        filepath = "{}/{}".format(self.pathname, filename)
+        if os.path.exists(filepath):
+            user = json.loads(open(filepath).read())
+        else:
+            user = None
+            for user_file in glob.glob("{}/user.*".format(self.pathname)):
+                user = json.loads(open(user_file).read())
+                if 'system' in user and user['system'] == "true":
+                    break
+                user = None
+            if not user:
+                # No system user
+                log.error("No system user for radosgw found")
+                return
+        self.credentials['access_key'] = user['keys'][0]['access_key']
+        self.credentials['secret_key'] = user['keys'][0]['secret_key']
+
+    def _urls(self):
+        """
+        Check for user defined endpoint; otherwise, return list of gateways as
+        urls.
+        """
+        search = "I@cluster:{}".format(self.cluster)
+        pillar_util = salt.utils.master.MasterPillarUtil(search, "compound",
+                                                         use_cached_grains=True,
+                                                         grains_fallback=False,
+                                                         opts='__opts__')
+        cached = pillar_util.get_minion_pillar()
+        for minion in cached:
+            if 'rgw_endpoint' in cached[minion]:
+                self.credentials['urls'].append(cached[minion]['rgw_endpoint'])
+                return
+
+        for client_file in glob.glob("{}/client.*".format(self.pathname)):
+            port = None
+            with open(client_file, 'r') as rgw_file:
+                for line in rgw_file:
+                    if 'port=' in line:
+                        match = re.search(r'port=(\d+)', line)
```

Comment:
this should cover the common use case for http, for https this will be say `443s` or some decimal succeeding s , also there is the case of port not appearing in the config, which defaults to civetweb port=7480 by ref http://docs.ceph.com/docs/master/install/install-ceph-gateway/#using-ssl-with-civetweb